techniques/system/serverRoles/1.0/logrotate-check.st

```diff
@@ -0,0 +1,67 @@
+# Must be run after the others components
+bundle common rudder_syslog {
+  vars:
+    any::
+      "user_name"        string => "root";
+      "group_name"       string => "root";
+      "file_mode" string => "640";
+
+      "service_name"     string => "rsyslog";
+      "rotation_method"  string => "reload";
+    ubuntu::
+      "user_name"        string => "syslog";
+    debian::
+      "group_name"       string => "adm";
+      "rotation_method"  string => "rotate";
+    SuSE::
+      "service_name"     string => "syslog";
+}
+
+bundle agent root_logrotate_check
+{
+  vars:
+    !redhat::
+      "reports_postrotate" string => "service ${rudder_syslog.service_name} ${rudder_syslog.rotation_method} > /dev/null || true";
+      "slapd_postrotate"   string => "service ${rudder_syslog.service_name} ${rudder_syslog.rotation_method} > /dev/null || true";
+    redhat::
+      "reports_postrotate" string => "/bin/kill -HUP `cat /var/run/syslogd.pid 2> /dev/null` 2> /dev/null || true";
+      "slapd_postrotate"   string => "/bin/kill -HUP `cat /var/run/syslogd.pid 2> /dev/null` 2> /dev/null || true";
+
+    pass1.!duration_configured::
+      "logrotate_duration" string => "30";
+    pass1.duration_configured::
+      "logrotate_duration" string => "${rudder_parameters.log_duration}";
+
+    pass1::
+      "logs_data"            data => parsejson('{
```

Comment:
All the template logic has been moved from the template to the technique code. The resulting template is much more readable.
Still, this is mustache templating and may lead to empty fields if we do not test it properly